proto/spire/next/api/entry/v1/entry.proto

```diff
@@ -37,47 +38,85 @@ message ListEntriesRequest {
         string by_spiffe_id = 3;
     }
 
-    // Filters the entries returned by the list operation.
+    // Filters the entries returned in the response.
+    //
+    // TODO: How to document optional fields?
     Filter filter = 1;
 
-    // The maximum number of items to return.
-    int32 page_size = 2;
+    // Masks the entry fields returned in the response.
+    spire.types.EntryMask mask = 2;
+
+    // The maximum number of entries to return.
+    int32 page_size = 3;
 
-    // The next_page_token value returned from a previous List request, if any.
-    string page_token = 3;
+    // The `next_page_token` value obtained from the previous `ListEntriesResponse` message, if any.
+    string page_token = 4;
 }
 
 message ListEntriesResponse {
+    // The list of entries.
     repeated spire.types.Entry entries = 1;
 
+    // The token to use as the value for `page_token` in the next `ListEntriesRequest`.
     string next_page_token = 2;
 }
 
 message GetEntryRequest {
+    // ID of the entry to get.
     string id = 1;
+
+    // Masks the entry fields returned in the response.
+    spire.types.EntryMask mask = 2;
 }
 
+// TODO: Do we never need a response message? CreateEntryIfNotExists needed
+// one to tell the caller whether the record was preexisting or not.
```

Comment:
I stopped short of adding a new RPC CreateEntryIfNotExists...

Probably don't want to return success on Create, with an existing entry, where the caller would be responsible for identifying the differences (if any)

Reply:
Ugg, the more we talk about entry creation behavior, the more i'm convinced `entry` is the wrong abstraction (at least at an API level).

If we stick with it, I think this is a place where we bite the bullet and make an exception against uniformity for this RPC and return a request message with an entry and boolean.

Reply:
The alternative is to have a `CreateIfNotExists` RPC here the same as we do in Registration API. It's not great, because I suspect more folks will use that then `Create`, but at least it's consistent.